### ISSUE TITLE
Return proper component foreground and background values

### ISSUE THREAD

**Issue by mgarin** [mgarin/weblaf#428]
Right now component foreground and background might return `null` values since the code that installed them was removed as values are not necessary for proper L&F functioning. That works fine for WebLaF but some 3rd-party libraries might not be ready for `null` values in fields like foreground and background and simply crash upon encountering such values.

I will add a temporary fix to setup dummy values into those fields from L&F, but it is only a workaround since returned values are incorrect. Real colors are kept within decoration and its contents and should probably be returned instead of the dummy values.

Also same thing might be applied to font later on as well.

**Comment by mgarin** [comment]
Maybe foreground should be set from an `AbstractTextContent` implementation that has baseline upon its activation? That would look something like this:
```java

    @Override
    public void activate ( final E c, final D d )
    {
        // Performing default actions
        super.activate ( c, d );

        // Setup foreground upon activation
        // This is a simple workaround to keep component foreground up-to-date with baseline text
        // Foreground is only updated in case component has UIResource instance of foreground
        if ( hasBaseline ( c, d ) && SwingUtils.isUIResource ( c.getForeground () ) )
        {
            final Color color = getColor ( c, d );
            if ( color != null )
            {
                c.setForeground ( new ColorUIResource ( color ) );
            }
        }
    }
```
Previous value can be kept and restored upon deactivation.

**Comment by mgarin** [comment]
For v1.2.9 version I will put common background/foreground values taken from defaults as a temporary solution. I will come back and address this later on in v1.3.0 to return actual used values.

**Comment by mgarin** [comment]
Temporary UI-level workaround have been added for all components with fully-custom UI class (not using Basic-UI implementations). 

Some custom components might still be missing some of the properties. If you encounter such cases simply report them here and I will add appropriate fixes.

**Comment by mgarin** [comment]
This actually have been already resolved in v1.2.10, all components should now correctly return background/foreground values as long as they are using one. It is specified in styles by default and is used directly by decoration elements, but can also be overridden for some if modified in those elements directly.
